### DISCUSSION TITLE
Ender 3 V2 4.2.2 cooling down after print - Don't want it, but can't change it?

### DISCUSSION THREAD

**Discussion by Shampoo1014** [mriscoc/Ender3V2S1#493] [Q&A]
Hey everyone, 
Just tried flashing the latest version of this firmware, which I am already loving!
But we have had an issue with multiple different firmwares now and was wondering whether anyone could shed any light on it: 
Our Ender 3 V2s cool down after a print has completed, even if we have removed _all_ cooldown commands (in the form of M104 Sxxx and M140 Sxxx). 
Running the exact same gcode on our Ender 3 Pros gives us the result we want, whereby the printer stays hot for the next print. We have scoured the firmware and cannot seem to find any mention to this forced cooldown command and it seems that perhaps it may have something to do with the LCD firmware? 
Does anyone have any idea how to remove this feature, or perhaps a good work around?

The goal is to have all of our printers on Octoprint (printers referred to here are running off SD cards for now) and our other Octoprint instances handle starting and ending scripts in order to run multiple gcodes over different printers. 

Thanks for the amazing firmware @mriscoc !

**Comment by mriscoc** [comment]
That is a security measure, you can disable it editing the dwin.cpp file:
```cpp
// Ended print job
void DWIN_Print_Finished() {
  TERN_(DEBUG_DWIN, SERIAL_ECHOLNPGM("DWIN_Print_Finished"));
  TERN_(POWER_LOSS_RECOVERY, if (card.isPrinting()) recovery.cancel());
  HMI_flag.pause_flag = false;
  wait_for_heatup = false;
  planner.finish_and_disable();
  thermalManager.cooldown();
  if (!HMI_flag.config_flag) Goto_PrintDone(); else HMI_flag.config_flag = false;
}
```

Comment out the line:

```cpp
  thermalManager.cooldown();
```

**Reply by Shampoo1014** [reply]
Thanks so much for the reply :) 
I have not edited dwin.cpp before, would we have to recompile and reflash firmware from scratch or would this just be the firmware for the LCD? 

Thanks so much for your time!

**Reply by mriscoc** [reply]
You don't need to flash you LCD. Only compile the firmware with your changes and flash the main control board of the printer.
